### ISSUE TITLE
conftest.py files in source directory creating problems

### ISSUE THREAD

**Issue by nickhand** [rainwoodman/runtests#14]
If you pass arguments to `runtests.py` that correspond to specific test files or directories to run, and those directories have their own `conftest.py` file, then the code will break. This happens because pytest loads those conftest files initially and then gets confused when the conftest files have moved to the build directory at runtime. The purpose of loading the conftest files initially (i.e., when creating the `config`object) is to see if they manipulate the command-line interface I think.

I think there are a couple of feasible options:

1. Set the `noconftest` flag when parsing the command-line so no conftest files are loaded initially. Then reset this flag so they are loaded with the proper path when the tests are run. 
2. Deprecate the `*args` command line option. If a specific test or module of tests is desired, this can still be specified as `-k test_module`. This seems to avoid all of the issues. So instead of doing

```
python runtests.py pyRSD/tests/test_derivatives 
```

do 

```
python runtests.py -k test_derivatives 
```

The first will be broken if `conftest.py` exists in the `pyRSD/tests/test_derivatives` directory. It's also seems a little weird to me to specify tests in the source directory when actually things are being run from the `build/` directory.

**Comment by rainwoodman** [comment]
I dislike the second proposal. 

- The tests are not necessarily subpackages (they don't need to carry a `__init__.py` file in the directory. It is merely a collection of test modules.

- In the original `-t` argument form, it stood for the test file to run. The relative path of the filename is sufficiently unique to determine the test suite. I always add a .py extension to make it a proper filename; it works with tab completion too.

What do you need `conftest.py` for? Are you importing confftest.py in the test suites? I suspect you are not supposed to do that. Is there still an issue if pytest is the only one that imports conftest.py?

**Comment by rainwoodman** [comment]
I reproduced this; the existence of conftest.py breaks it.

Is there anything that you absolutely need conftest for?

Can you move the configuration into runtests.py instead? For very large projects there may be some use having different conftest in different test directories. But for a properly decoupled project all tests shall probably share the same configuration..

**Comment by nickhand** [comment]
I am using a conftest.py to load a fixtures for parts of the test suite. In this case, it's loading a RSD model that takes a long time to initialize, which is then shared among multiple tests. I think this is the right use of a conftest file in the pytest framework...not sure how to accomplish that otherwise. 

If we go with option 1 above, the only functionality that is lost would be manipulating the command-line via user-defined plugins in conftest files, which is likely less common that then use of conftests in general. I think they are relatively common in the pytest framework, i.e., http://stackoverflow.com/questions/34466027/in-py-test-what-is-the-use-of-conftest-py-files

**Comment by rainwoodman** [comment]
pytest does a pre-parsing stage that does not actually invoke any
conftests. I wonder if we can make use of that??

**Comment by nickhand** [comment]
closing for now with the fix from 748b103
